Comments/ClientApp/src/components/CommentList/CommentList.js

```diff
@@ -104,25 +104,36 @@ export class CommentList extends Component<PropsType, StateType> {
 		);
 
 		if (preloadedCommentsForCurrentUser) {
-			let allowComments = preloadedCommentsForCurrentUser.consultationState.consultationIsOpen && !preloadedCommentsForCurrentUser.consultationState.submittedDate;
+			const { comments, questions, consultationState } = preloadedCommentsForCurrentUser;
+			const {
+				consultationIsOpen,
+				submittedDate,
+				shouldShowDrawer,
+				shouldShowCommentsTab,
+				shouldShowQuestionsTab,
+				endDate,
+			} = consultationState;
+
+			const allowComments = consultationIsOpen && !submittedDate;
+			const shouldShowCommentsTabOverride = !shouldShowCommentsTab && !shouldShowQuestionsTab ? true : shouldShowCommentsTab;
 
 			this.state = {
-				comments: preloadedCommentsForCurrentUser.comments,
-				questions: preloadedCommentsForCurrentUser.questions,
+				comments,
+				questions,
 				loading: false,
-				allowComments: allowComments,
+				allowComments,
 				error: "",
 				initialDataLoaded: true,
-				viewComments: preloadedCommentsForCurrentUser.consultationState.shouldShowCommentsTab,
-				shouldShowDrawer: preloadedCommentsForCurrentUser.consultationState.shouldShowDrawer,
-				shouldShowCommentsTab: preloadedCommentsForCurrentUser.consultationState.shouldShowCommentsTab,
-				shouldShowQuestionsTab: preloadedCommentsForCurrentUser.consultationState.shouldShowQuestionsTab,
-				drawerOpen: false,
+				viewComments: shouldShowCommentsTabOverride, // why is this not set below?
```

Comment:
i don't really know what viewComments is for here. do you think we could get rid of it? there's a few references to it below, but presumably they could just use shouldShowCommentsTab